templates/openharmony/entry/src/main/ets/workers/cocos_worker.ts

```diff
@@ -54,22 +66,40 @@ parentPort.onmessage = function(e) {
             }).catch(e => {
                 console.error('launch CC engien failed');
             });
+           // @ts-ignore
+            window.oh.postMessage = nativeContext.postMessage;
+           // @ts-ignore
+            window.oh.postSyncMessage = nativeContext.postSyncMessage;
             renderContext.nativeEngineStart();
             break;
         case "onTextInput":
-            nativeEditBox.onTextChange(data.data);
+            nativeEditBox.onTextChange(msg.param);
             break;
         case "onComplete":
-            nativeEditBox.onComplete(data.data);
+            nativeEditBox.onComplete(msg.param);
             break;
         case "onPageBegin":
-            nativeWebView.shouldStartLoading(data.data.viewTag, data.data.url);
+            nativeWebView.shouldStartLoading(msg.param.viewTag, msg.param.url);
             break;
         case "onPageEnd":
-            nativeWebView.finishLoading(data.data.viewTag, data.data.url);
+            nativeWebView.finishLoading(msg.param.viewTag, msg.param.url);
             break;
         case "onErrorReceive":
-            nativeWebView.failLoading(data.data.viewTag, data.data.url);
+            nativeWebView.failLoading(msg.param.viewTag, msg.param.url);
+            break;
+        case "onVideoEvent":
+            // For native Implementation
+//            if(msg.param.args != undefined || msg.param.args != null) {
+//                //nativeVideo.onEvents(msg.param.videoTag, msg.param.videoEvent, msg.param.args);
+//            } else {
+//                //nativeVideo.onEvents(msg.param.videoTag, msg.param.videoEvent);
+//            }
```

Comment:
Can delete these codes?

Reply:
I want to keep the native implementation. Not sure if ts implementation will have problems later.

And in the future We may use the video texture feature, not sure if I will do the implementation in native later.